### PR TITLE
REF: use BlockManager.apply for cython_agg_blocks, apply_blockwise

### LINKED ISSUE #34714: REF: BlockManager.apply_allow_failures
cc @mroeschke w/r/t rolling `._apply`, @WillAyd w/r/t `_cython_agg_blocks`, and anyone else w/r/t `DataFrame._reduce`with `numeric_only=None`,  `apply.FrameApply` with `ignore_failures=True`.

All of these do something along the lines of:

```
results = []
exclude = []
for i, block in enumerate(mgr.blocks):
    try:
          res = func(block.values)
          results.append(res)
    except:
          exclude.append(i)

out = reconstruct(results, exclude)
```

Two things should be done with this pattern:
1) Deprecate it, since it is a disproportionate maintainence burden
2) implement something like `BlockManager.apply_with_ignore_failures` that would resemble BlockManager.apply but with the try/except logic\*

\* We _could_ add that logic into BlockManager.apply, but BM.apply assumes the output has the same shape as the original frame, which I don't think holds for the cases mentioned above, so that would be a little more invasive than just adding a try/except.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #34714
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -349,14 +349,20 @@ def operate_blockwise(self, other: "BlockManager", array_op) -> "BlockManager":
...
+        self: T, f, align_keys=None, ignore_failures: bool = False, **kwargs
+    ) -> T:
         """
         Iterate over the blocks, collect and create a new BlockManager.
 
         Parameters
         ----------
         f : str or callable
             Name of the Block method to apply.
+        align_keys: List[str] or None, default None
```
Comment (MarcoGorelli) [inline pandas/core/internals/managers.py:362]: should `align_keys` be typed in the signature too?

Comment (jbrockmendel) [reply pandas/core/internals/managers.py:362]: good idea, updated

**Review by jreback — APPROVED** [review]
lgtm. not sure when last merged master?

**Comment by jbrockmendel** [comment]
rebased+green

**Pull request merged**